v2/gcs-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/GCSToSourceDbWithoutReaderIT.java

```diff
@@ -0,0 +1,225 @@
+    for (String d : ddls) {
+      if (!d.isBlank()) {
+        spannerResourceManager.executeDdlStatement(d);
+      }
+    }
+    return spannerResourceManager;
+  }
+
+  private SpannerResourceManager createSpannerMetadataDatabase() throws IOException {
+    SpannerResourceManager spannerMetadataResourceManager =
+        SpannerResourceManager.builder("rr-meta-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String dummy = "create table t1(id INT64 ) primary key(id)";
+    spannerMetadataResourceManager.executeDdlStatement(dummy);
+    return spannerMetadataResourceManager;
+  }
+
+  private void createMySQLSchema(CustomMySQLResourceManager jdbcResourceManager) {
+    HashMap<String, String> columns = new HashMap<>();
+    columns.put("id", "INT NOT NULL");
+    columns.put("name", "VARCHAR(25)");
+    JDBCResourceManager.JDBCSchema schema = new JDBCResourceManager.JDBCSchema(columns, "id");
+
+    jdbcResourceManager.createTable(TABLE, schema);
+  }
+
+  private void launchWriterDataflowJob() throws IOException {
+    Map<String, String> params =
+        new HashMap<>() {
+          {
+            put("sessionFilePath", getGcsPath("input/session.json"));
+            put("spannerProjectId", PROJECT);
+            put("metadataDatabase", spannerMetadataResourceManager.getDatabaseId());
+            put("metadataInstance", spannerMetadataResourceManager.getInstanceId());
+            put("sourceShardsFilePath", getGcsPath("input/shard.json"));
+            put("runIdentifier", "run1");
+            put("GCSInputDirectoryPath", getGcsPath("cdc"));
+            put("startTimestamp", "2024-05-13T08:43:10.000Z");
+            put("windowDuration", "10s");
+          }
+        };
+    String jobName = PipelineUtils.createJobName(testName);
+    LaunchConfig.Builder options = LaunchConfig.builder(jobName, specPath);
+    options.setParameters(params);
+    // Run
+    jobInfo = launchTemplate(options, false);
+  }
+
+  private void createAndUploadShardConfigToGcs(
+      GcsResourceManager gcsResourceManager, CustomMySQLResourceManager jdbcResourceManager)
+      throws IOException {
+    Shard shard = new Shard();
+    shard.setLogicalShardId("Shard1");
+    shard.setUser(jdbcResourceManager.getUsername());
+    shard.setHost(jdbcResourceManager.getHost());
+    shard.setPassword(jdbcResourceManager.getPassword());
+    shard.setPort(String.valueOf(jdbcResourceManager.getPort()));
+    shard.setDbName(jdbcResourceManager.getDatabaseName());
+    JsonObject jsObj = (JsonObject) new Gson().toJsonTree(shard).getAsJsonObject();
+    jsObj.remove("secretManagerUri"); // remove field secretManagerUri
+    JsonArray ja = new JsonArray();
+    ja.add(jsObj);
+    String shardFileContents = ja.toString();
+    LOG.info("Shard file contents: {}", shardFileContents);
+    gcsResourceManager.createArtifact("input/shard.json", shardFileContents);
+  }
+}
```

Comment:
Are there more expected ITs for `GCSToSourcdDB*`? If yes, we should put this in a `TestBase` class similar to our approach in `DataStreamToSpanner`

Reply:
Yes. I'll move out the common methods in next PRs along with the actual test cases.